pex/cli/commands/cache/command.py

```diff
@@ -0,0 +1,391 @@
+# Copyright 2024 Pex project contributors.
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, print_function
+
+import os
+from argparse import Action, ArgumentError, _ActionsContainer
+from datetime import datetime
+
+from pex.cache import access as cache_access
+from pex.cache.dirs import CacheDir
+from pex.cli.command import BuildTimeCommand
+from pex.cli.commands.cache.bytes import ByteAmount, ByteUnits
+from pex.cli.commands.cache.du import DiskUsage
+from pex.commands.command import OutputMixin
+from pex.common import pluralize, safe_rmtree
+from pex.exceptions import reportable_unexpected_error_msg
+from pex.jobs import iter_map_parallel, map_parallel
+from pex.orderedset import OrderedSet
+from pex.result import Error, Ok, Result
+from pex.typing import TYPE_CHECKING
+from pex.variables import ENV
+
+if TYPE_CHECKING:
+    from typing import IO, Dict, Iterable, List, Optional, Tuple, Union
+
+
+class HandleAmountAction(Action):
+    def __init__(self, *args, **kwargs):
+        kwargs["nargs"] = 0
+        super(HandleAmountAction, self).__init__(*args, **kwargs)
+
+    def __call__(self, parser, namespace, value, option_str=None):
+        if option_str in ("-H", "--human-readable"):
+            amount_func = ByteAmount.human_readable
+        elif option_str.startswith("--"):
+            amount_func = ByteAmount.for_unit(ByteUnits.for_value(option_str[2:]))
+        else:
+            raise ArgumentError(
+                argument=self,
+                message=reportable_unexpected_error_msg(
+                    "The HandleAmountAction was used in an unexpected place."
+                ),
+            )
+        setattr(namespace, self.dest, amount_func)
+
+
+class Cache(OutputMixin, BuildTimeCommand):
+    """Interact with the Pex cache."""
+
+    @staticmethod
+    def _add_amount_argument(parser):
+        # type: (_ActionsContainer) -> None
+
+        parser.add_argument(
+            "-H",
+            "--human-readable",
+            *["--{unit}".format(unit=unit.value) for unit in ByteUnits.values()],
+            dest="amount_func",
+            action=HandleAmountAction,
+            default=ByteAmount.bytes,
+            help=(
+                "How to display disk usage amounts; defaults to bytes. The -H / --human-readable "
+                "options display amounts in a human readable way and each of the other unit "
+                "options displays amounts with that specified unit."
+            )
+        )
+
+    @classmethod
+    def _add_info_arguments(cls, parser):
+        # type: (_ActionsContainer) -> None
+
+        cls._add_amount_argument(parser)
+        parser.add_argument(
+            "-S",
+            "--sort-size",
+            dest="sort_by_size",
+            action="store_true",
+            help=(
+                "Don't actually purge cache entries; instead, perform a dry run that just prints "
```

Comment:
The help strings here and for `--reverse` need updating.

Reply:
Fixed.